Userland/Libraries/LibMedia/Manip.cpp

```diff
@@ -0,0 +1,35 @@
+#include <LibMedia/Manip.h>
+#include <iostream>
+
+namespace Media {
+
+AK::String u32_to_fourcc(const u32 val)
+{
+    // Little-endian
+    const char fcc[5] = { static_cast<char>(val),
+        static_cast<char>(val >> 8),
+        static_cast<char>(val >> 16),
+        static_cast<char>(val >> 24),
+        '\0' };
+    return AK::String(fcc);
+}
+
+u16 bytes_to_u16le(const ByteBuffer& bytes)
```

Comment:
A more serenity-like way to decode bytes in a particular byte order from a memory stream would be to use AK::InputMemoryStream and AK::LittleEndian/AK::BigEndian.

LibAudio and LibCompress have some decent examples on doing this.

Reply:
Thanks for pointing that one out. Are AK types ifaces fixed? This one doesn't do what is needed.

As for the fn in question, i'd be keen to keep this as it allows flexibility within the various types. A bitstreamreader is already being used to read across chunk/boxes boundaries.

Reply:
I'm not sure what you mean. You can extract a u16 little endian from a ByteBuffer easily like so:

```c++
InputMemoryStream my_stream(my_bytes);
LittleEndian<u16> my_val = {};

stream >> my_val;

if (stream.handle_any_error())
    return false; // nope, invalid input byte stream, unable to load your data.
```

If you're referring to your BitStreamReader under `reader`, there's an existing class for that in AK, AK/BitStream.h. AK::InputBitStream wraps an existing InputStream (such as a InputFileStreamor InputMemoryStream) and allows you to read arbitrary bit offsets. This is something the LibAuido/FlacLoader uses already. It seems that the VP9 decoder in LibVideo invented its own way of doing the same thing, it should probably be changed to use AK::Bitstream 🤔 .

In regards to AK type interfaces being fixed or not, no, you're free to add functionality to them in their own commits. Though it's nice to add tests for the new features to AK classes in Tests/AK.

Reply:
I'll use that Bitstreamreader instead for reading from the file. I'll have a go at that method of extracting types but I have reservations as I can foresee it ballooning code. We'll see.